x-pack/plugins/apm/server/lib/alerts/register_transaction_duration_anomaly_alert_type.ts

```diff
@@ -81,6 +81,11 @@ export function registerTransactionDurationAnomalyAlertType({
         anomalyDetectors,
         alertParams.environment
       );
+
+      if (mlJobIds.length === 0) {
+        return {};
+      }
```

Comment:
Was this an unrelated bug you ran into?

Reply:
Hmm, debatable. Previously this function threw an error, so I now have an extra check to prevent a useless search request below. The bug could be that if you create an anomaly alert without any active ML jobs, or if you stop those jobs later, that the executor would throw an error. I don't think the user would know the difference, just noise in the logs.